### DISCUSSION TITLE
Why cant we use negative sampling for sequential recommender?

### DISCUSSION THREAD

**Discussion by Jeriousman** [RUCAIBox/RecBole#1595] [General]
In Recbole now, It seems that sequential recommenders are not utilizing BPR loss with negative sampling? Why is that the case?

Generating negative sequence shouldnt help? Even BERT4Rec official paper noted that they did negative sampling but BERT4Rec in Recbole doesnt seem to support negative sampling as well. 

Can someone please explain why Recbole isnt doing so?

**Comment by AoiDragon** [comment]
Hello @Jeriousman , sequential recommenders in RecBole support BPR loss with negative sampling. You can set `loss_type` to `BPR` and specify `train_neg_sample_args`.  For more detailed information, you can refer to our API doc. [Here](https://recbole.io/docs/user_guide/model/sequential/gru4rec.html) is an example of GRU4Rec.
